language-design.md

```diff
@@ -74,7 +230,27 @@ implementation choices from users?
 
 **Response**
 
-
+The choice of implementation for a DSL strongly affects its users. If we choose
+to implement an internal DSL, then users can have an easier time learning the
+language, because users may already be familiar with the host language. Even if
+they are not, the host language will have documentation, so it will likely be
+easier to pick up the DSL than if the DSL was a totally new language. On the
+other hand, if we implement our DSL as external, there is the advantage that we
+can choose syntax that makes sense to experts in our chosen domain. If those
+experts are our users, that will help them.
+
+In some cases it is not possible to hide implementation choices from users. For
+example, users will be able to tell whether our language is internal or
+external. However, whenever we can, it would be advisable to hide as much as
+possible. For example, if our language has a `sort` keyword, we do not need to
+tell the user more than how fast it is guaranteed to run. This is a principle
```

Comment:
But if the user knows they will be mostly sorting lists that are already mostly sorted, wouldn't it be nice to give them the power to choose their algorithm without having to roll their own? I agree that it's not necessary to tell the user but it might be nice to give them options. Maybe there could be a `sort` function but also `mergesort` and `insertionshort` functions